Split disable alerts all into disable low and high

JohanDegraeve opened the pull request:
Snooze Screen : split disable alerts into 'disable all', 'disable low' and 'disable high' - also delete active bg alerts that fall within range of disabled alerts.
- The screen now has three buttons on the bottom : one to just disable low bg alerts, one for the high bg alerts and one for all alerts
- Disable and re-enable all alerts works as before, when disabling all, the two other buttons disappear. 
- When re-enabling all, the two others buttons re-appear in the screen, and also bg low and high alerts will be re-enabled, whatever the status was before
- If high alert exists, and disable high or all is done, then the existing high alert will be deleted.
- If low alert exists, and disable low or all is done, then the existing low alert will be deleted
- in the home screen, the warning messages is adapted according to the alerts being disabled : "All alerts, high alerts or low alerts"

(ignore the commit i did related to chaning warm up period of sensor - this I do during testing so I don't need to wait 2 hours)

JohanDegraeve committed:
After starting new sensor, wait only 10 minutes in stead of 2 hours

This is interesting in case the app is just installed but a sensor already exists for longer time. In that case there's no need to wait for 2 hours. 10 minutes is the minimum the app needs to calculate values.

JohanDegraeve committed:
Merge branch 'master' into sensor-warm-up-only-10-minutes

JohanDegraeve committed:
Snooze Screen : split disable alerts into 'disable all', 'disable low' and 'disable high' - also delete active bg alerts that fall within range of disabled alerts.

- The screen now has three buttons on the bottom : one to just disable low bg alerts, one for the high bg alerts and one for all alerts
- Disable and re-enable all alerts works as before, when disabling all, the two other buttons disappear.
- When re-enabling all, the two others buttons re-appear in the screen, and also bg low and high alerts will be re-enabled, whatever the status was before
- If high alert exists, and disable high or all is done, then the existing high alert will be deleted
- If low alert exists, and disable low or all is done, then the existing low alert will be deleted
- in the home screen, the warning messages is adapted according to the alerts being disabled : "All alerts, high alerts or low alerts"

JohanDegraeve committed:
reset warm up time to 10 minutes

tzachi-dar commented:
On first glance looks good. I'll give it a more thorough look in the coming days.

tzachi-dar commented on app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java:
```diff
@@ -767,6 +767,10 @@ public static void checkForRisingAllert(Context context) {
             Log.i("NOTIFICATIONS", "checkForRisingAllert: Notifications are currently disabled!!");
             return;
         }
+        if(prefs.getLong("high_alerts_disabled_until", 0) > new Date().getTime()){
```
I don't think this is really related to the low high alert. Also this alerts is only fired once so it does not bother too much in any case.

JohanDegraeve replied:
ok I will remove that check, rise alert indeed is something different than high alert

tzachi-dar commented on app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java:
```diff
@@ -801,6 +805,10 @@ public static void checkForDropAllert(Context context) {
             return;
         }
+        if(prefs.getLong("low_alerts_disabled_until", 0) > new Date().getTime()){
+            Log.d("NOTIFICATIONS", "checkForDropAllert: Low alerts are currently disabled!!");
```
Same as above.

JohanDegraeve replied:
ok I will remove that check, drop alert indeed is something different than low alert

mgranberry commented on app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java:
```diff
@@ -125,6 +129,16 @@ public String getMenuName() {
     public void addListenerOnButton() {
         buttonSnooze = (Button)findViewById(R.id.button_snooze);
+		
+		//low alerts
```
The whitespace appears to be off here.

JohanDegraeve replied:
will remove that

JohanDegraeve replied:
the white space looks good in my source file

tzachi-dar commented:
One last request:
Can you remove alerts_disabled_until from the code?
keep the 3 buttons but once the all one is pressed, simply call the disable low and disable high.

Thanks
Tzachi

JohanDegraeve commented:
in the current implementation alerts_disabled_until disables really all : low , high , drop , rise, calibration alerts (if i'm not mistaking)

Do you propose then that only low and high alerts can be disabled ?

tzachi-dar commented:
My mistake then, I was not aware that disable all alerts really disables all of them... I thought that it was disabling only the high/low.
So, ignore my comment about  alerts_disabled_until.
Tzachi

tzachi-dar commented:
By the way, not really related to this checkin but just FYI.
If you want to have a faked sensor started immediately you can simply start it with a time from the past, no need to change anything in code.
Also, if you want to have a system for testing that gets bg going up and down, you can simply switch to wifi-wixel mode.
Than in code, replace the function readDataFake as readData. it will than give you a system that has readings going up and down.

JohanDegraeve committed:
Drop and rise alert are not disabled when chosing low or high disable.

-  They are disabled only in case the user choses "Disable All alerts", as was the case before

JohanDegraeve commented:
thanks for this info - I pshed the changes

AdrianLxM commented:
While you/we are at this part of xDrip: How about an "un-snooze" function? Or maybe simpler: adding "0 minutes" to the snooze time selection spinner?

JohanDegraeve commented:
Could be achieved with this new change by first "disable all alerts" (or disable low or high alerts), and then "re-enable all alerts".
When disabling the alerts, the active alerts are deleted (that's also a change I did)

Maybe not very clear for the end-user/ could be useful to have a "help" button which explains things a little bit

AdrianLxM commented:
@JohanDegraeve, I like that behaviour. Works for me :)

tzachi-dar commented:
Thanks for this patch.

AdrianLxM commented:
Before merging this I would wait for Marshmallow. It will cause merge conflicts that are easier resolved the other way around. And making xDrip usable for Marshmallow seems to be more urgent to me.

tzachi-dar commented:
Looks a bit too late for that (already merged).

AdrianLxM commented:
may I revert that?

tzachi-dar commented:
when do we expect to have marshmellow merged?
Can we rebase that over the master?

AdrianLxM commented:
I'd say @mgranberry has the lead on that ;)

mgranberry commented:
I just merged master back into the marshmallow branch.  No worries.
